src/main/java/com/coreos/jetcd/EtcdClient.java

```diff
@@ -14,15 +15,20 @@
     private final ManagedChannel channel;
 
     private final EtcdKV kvClient;
+    private final EtcdLease leaseClient;
 
     public EtcdClient(ManagedChannelBuilder<?> channelBuilder, EtcdClientBuilder builder) {
```

Comment:
I probably missed bunch of changes here. But I feel it is a bit strange to have circle dependency between builder and client.

Ideally it would be:
- Builder builds a client
- Builder constructs a configuration that based on application's build methods
- Builder passes configuration to client rather than pass builder itself to the client.